<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to 'gggenes' • gggenes</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to 'gggenes'">
<meta property="og:description" content="gggenes">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/introduction-to-gggenes.html">Introduction to 'gggenes'</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wilkox/gggenes/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="introduction-to-gggenes_files/header-attrs-2.5/header-attrs.js"></script><link href="introduction-to-gggenes_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="introduction-to-gggenes_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to ‘gggenes’</h1>
                        <h4 class="author">David Wilkins</h4>
            
            <h4 class="date">2020-12-10</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wilkox/gggenes/blob/master/vignettes/introduction-to-gggenes.Rmd"><code>vignettes/introduction-to-gggenes.Rmd</code></a></small>
      <div class="hidden name"><code>introduction-to-gggenes.Rmd</code></div>

    </div>

    
    
<p>‘gggenes’ is a (quite small) set of tools for drawing gene arrow maps with ‘ggplot2.’</p>
<div id="drawing-gene-arrows-with-geom_gene_arrow" class="section level2">
<h2 class="hasAnchor">
<a href="#drawing-gene-arrows-with-geom_gene_arrow" class="anchor"></a>Drawing gene arrows with <code>geom_gene_arrow()</code>
</h2>
<p><code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> is a ‘ggplot2’ geom that represents genes with arrows. The start and end locations of the genes within their molecule(s) are mapped to the <code>xmin</code> and <code>xmax</code> aesthetics respectively. These start and end locations are used to determine the directions in which the arrows point. The <code>y</code> aesthetic must be mapped to the molecule(s). If you are drawing more than one molecule, and the numerical locations of the genes are not similar across molecules, you almost certainly want to facet the plot with <code>scales = "free"</code> to avoid drawing ridiculously large molecules with ridiculously tiny genes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkox.org/gggenes/">gggenes</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/geom_gene_arrow-1.png" width="576"></p>
</div>
<div id="beautifying-the-plot-with-theme_genes" class="section level2">
<h2 class="hasAnchor">
<a href="#beautifying-the-plot-with-theme_genes" class="anchor"></a>Beautifying the plot with <code>theme_genes()</code>
</h2>
<p>Because the resulting plot can look cluttered, a ‘ggplot2’ theme <code><a href="../reference/theme_genes.html">theme_genes()</a></code> is provided with some sensible defaults.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/theme_genes-1.png" width="576"></p>
</div>
<div id="aligning-genes-across-facets-with-make_alignment_dummies" class="section level2">
<h2 class="hasAnchor">
<a href="#aligning-genes-across-facets-with-make_alignment_dummies" class="anchor"></a>Aligning genes across facets with <code>make_alignment_dummies()</code>
</h2>
<p>Often you will want a certain gene to be vertically aligned across the faceted molecules. <code><a href="../reference/make_alignment_dummies.html">make_alignment_dummies()</a></code> generates a set of ‘dummy’ genes that if added to the plot with <code><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">ggplot2::geom_blank()</a></code> will extend the range of each facet to visually align the selected gene across facets.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dummies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_alignment_dummies.html">make_alignment_dummies</a></span><span class="op">(</span>
  <span class="va">example_genes</span>,
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, id <span class="op">=</span> <span class="va">gene</span><span class="op">)</span>,
  on <span class="op">=</span> <span class="st">"genE"</span>
<span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">geom_blank</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dummies</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/make_alignment_dummies-1.png" width="576"></p>
</div>
<div id="labelling-genes-with-geom_gene_label" class="section level2">
<h2 class="hasAnchor">
<a href="#labelling-genes-with-geom_gene_label" class="anchor"></a>Labelling genes with <code>geom_gene_label()</code>
</h2>
<p>To label individual genes, provide a <code>label</code> aesthetic and use <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code>. <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code> uses the <a href="https://github.com/wilkox/ggfittext">‘ggfittext’</a> package to fit the label text inside the gene arrows; see the ‘ggfittext’ documentation for more details on how it resizes and reflows text to make it fit.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span>
                                            <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>, label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span>arrowhead_height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"mm"</span><span class="op">)</span>, arrowhead_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_label.html">geom_gene_label</a></span><span class="op">(</span>align <span class="op">=</span> <span class="st">"left"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_blank.html">geom_blank</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dummies</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/labelled_genes-1.png" width="576"></p>
</div>
<div id="reversing-gene-direction-with-the-optional-forward-aesthetic" class="section level2">
<h2 class="hasAnchor">
<a href="#reversing-gene-direction-with-the-optional-forward-aesthetic" class="anchor"></a>Reversing gene direction with the optional <code>forward</code> aesthetic</h2>
<p>By default, genes arrows are drawn pointing in the direction implied by <code>xmin</code> and <code>xmax</code>. You can override this with the optional <code>forward</code> aesthetic, which can be useful when the gene coordinates and orientation are encoded as separate variables.</p>
<p>If <code>forward</code> is TRUE (the default), or any value that coerces to TRUE such as 1, the gene will be drawn pointing in the implied direction, i.e. from <code>xmin</code> to <code>xmax</code>. If <code>forward</code> is FALSE, or any value that coerces to FALSE such as -1, the gene will be drawn in the reverse of this implied direction:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>, 
                          forward <span class="op">=</span> <span class="va">orientation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/reversing_direction-1.png" width="576"></p>
</div>
<div id="viewing-subgene-segments" class="section level2">
<h2 class="hasAnchor">
<a href="#viewing-subgene-segments" class="anchor"></a>Viewing subgene segments</h2>
<p>We can highlight subgene segments, such as protein domains or local alignments, using <code><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow()</a></code>.</p>
<p>This works similarly to <code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code>, but in addition to <code>xmin</code> and <code>xmax</code> (which determine the gene boundaries), we need the aesthetics <code>xsubmin</code> and <code>xsubmax</code> to determine the subgene boundaries. <code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> will produce pretty arrowheads, as long as <code>xmin &gt;= xsubmin</code> and <code>xmax &gt;= xsubmax</code> for all subgenes (subgenes that break gene boundaries will be skipped with a warning).</p>
<p>The suggested usage is to use <code><a href="../reference/geom_gene_arrow.html">geom_gene_arrow()</a></code> with no fill, and then add a subgene layer over this:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">molecule</span>, scales <span class="op">=</span> <span class="st">"free"</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">example_subgenes</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">molecule</span>, fill <span class="op">=</span> <span class="va">gene</span>,
        xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, alpha<span class="op">=</span><span class="fl">.7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/theme_genes.html">theme_genes</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/subgenes-1.png" width="576"></p>
<p>To label subgenes, we can use <code><a href="../reference/geom_subgene_label.html">geom_subgene_label()</a></code>, which works similarly to <code><a href="../reference/geom_gene_label.html">geom_gene_label()</a></code> with the major difference that it requires <code>xsubmin</code> and <code>xsubmax</code> aesthetics (not <code>xmin</code> and <code>xmax</code>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">example_genes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,
       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, y <span class="op">=</span> <span class="va">strand</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_arrow.html">geom_gene_arrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_gene_label.html">geom_gene_label</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_subgene_arrow.html">geom_subgene_arrow</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">example_subgenes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span>, fill <span class="op">=</span> <span class="va">subgene</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/geom_subgene_label.html">geom_subgene_label</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">example_subgenes</span>, <span class="va">molecule</span> <span class="op">==</span> <span class="st">"Genome4"</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op">==</span> <span class="st">"genA"</span><span class="op">)</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xsubmin <span class="op">=</span> <span class="va">from</span>, xsubmax <span class="op">=</span> <span class="va">to</span>, label <span class="op">=</span> <span class="va">subgene</span><span class="op">)</span>,
    min.size <span class="op">=</span> <span class="fl">0</span>
  <span class="op">)</span>
<span class="co">#&gt; Warning in asin(abs(rise)/abs(magnitude)): NaNs produced</span>

<span class="co">#&gt; Warning in asin(abs(rise)/abs(magnitude)): NaNs produced</span>

<span class="co">#&gt; Warning in asin(abs(rise)/abs(magnitude)): NaNs produced</span>

<span class="co">#&gt; Warning in asin(abs(rise)/abs(magnitude)): NaNs produced</span></code></pre></div>
<p><img src="introduction-to-gggenes_files/figure-html/subgene%20labels-1.png" width="576"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Wilkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
